File: 05-if-else/main.go
package main

import "fmt"

func getOptionalValue(check bool) *int {
	if check {
		v := 42
		return &v
	}
	return nil
}

func main() {
	// === 基本 if/else ===
	x := 10

	if x > 5 {
		fmt.Println("x 大于 5")
	} else if x > 0 {
		fmt.Println("x 大于 0 但不大于 5")
	} else {
		fmt.Println("x 小于等于 0")
	}

	// === Go 没有三元运算符，必须用 if/else 语句 ===
	var y int
	if x > 5 {
		y = 100
	} else {
		y = 200
	}
	fmt.Printf("y = %d\n", y)

	// === 同样，初始化变量也需要 if 语句 ===
	var label string
	if x%2 == 0 {
		label = "even"
	} else {
		label = "odd"
	}
	fmt.Printf("x is %s\n", label)

	// === Go 的 nil 检查（对比 Zig 的 optional 捕获） ===
	maybeVal := getOptionalValue(true)
	if maybeVal != nil {
		fmt.Printf("捕获到值: %d\n", *maybeVal)
	} else {
		fmt.Println("值为 nil")
	}

	noVal := getOptionalValue(false)
	if noVal != nil {
		fmt.Printf("捕获到值: %d\n", *noVal)
	} else {
		fmt.Println("值为 nil")
	}

	// === Go 特色：if 中声明变量 ===
	if val := getOptionalValue(true); val != nil {
		fmt.Printf("if 中声明并使用: %d\n", *val)
	}
	// val 在这里已经不可访问

	// === bool 运算（Go 用 && || ） ===
	a := true
	b := false
	if a && !b {
		fmt.Println("a 为 true 且 b 为 false")
	}
	if a || b {
		fmt.Println("a 或 b 至少一个为 true")
	}
}

File: 05-if-else/main.zig
const std = @import("std");
const print = std.debug.print;

fn getOptionalValue(check: bool) ?i32 {
    if (check) return 42;
    return null;
}

pub fn main() void {
    // === 基本 if/else ===
    const x: i32 = 10;

    if (x > 5) {
        print("x 大于 5\n", .{});
    } else if (x > 0) {
        print("x 大于 0 但不大于 5\n", .{});
    } else {
        print("x 小于等于 0\n", .{});
    }

    // === if 作为表达式（代替三元运算符） ===
    const y: i32 = if (x > 5) 100 else 200;
    print("y = {}\n", .{y});

    // === 用 if 表达式初始化变量 ===
    const label: []const u8 = if (x % 2 == 0) "even" else "odd";
    print("x is {s}\n", .{label});

    // === Optional 捕获：Zig 的特色功能 ===
    const maybe_val: ?i32 = getOptionalValue(true);

    // 如果 optional 有值，用 |val| 捕获
    if (maybe_val) |val| {
        print("捕获到值: {}\n", .{val});
    } else {
        print("值为 null\n", .{});
    }

    // 没有值的情况
    const no_val: ?i32 = getOptionalValue(false);
    if (no_val) |val| {
        print("捕获到值: {}\n", .{val});
    } else {
        print("值为 null\n", .{});
    }

    // === bool 运算 ===
    const a = true;
    const b = false;
    // Zig 用 and/or 关键字，不用 && ||
    if (a and !b) {
        print("a 为 true 且 b 为 false\n", .{});
    }
    if (a or b) {
        print("a 或 b 至少一个为 true\n", .{});
    }
}
